### PR TITLE
refactor(local-debug): update aad plugin to use localSettings for local debug

### PR DESCRIPTION
**Pull request by dooriya**
If multi-env feature flag is enabled, AAD plugin will use `localSettings.json` to read/write local debug related configuration. Otherwise, the local debug config logic remains the same as original.

### CODE REVIEW

File: packages/fx-core/src/common/localSettingsConstants.ts [hunk 1]
```diff
@@ -8,11 +8,11 @@ export const LocalSettingsTeamsAppKeys = Object.freeze({
 });
 
 export const LocalSettingsAuthKeys = Object.freeze({
-  AadClientId: "aadClientId",
-  AadClientSecret: "aadClientSecret",
-  AadObjectId: "aadObjectId",
-  AadOauth2PermissionScopeId: "aadOauth2PermissionScopeId",
-  AadApplicationIdUris: "aadApplicationIdUris",
+  AadClientId: "clientId",
```
Comment (swatDong) [inline packages/fx-core/src/common/localSettingsConstants.ts:11]: Just better style, align the key and value `ClientId: "clientId"`.

File: packages/fx-core/src/plugins/resource/aad/utils/configs.ts [hunk 2]
```diff
@@ -8,14 +8,84 @@ import { format, Formats } from "./format";
...
+import { getArmOutput, isArmSupportEnabled, isMultiEnvEnabled } from "../../../../common";
+import {
+  LocalSettingsBotKeys,
+  LocalSettingsFrontendKeys,
+} from "../../../../common/localSettingsConstants";
 
-function checkAndSaveConfig(context: PluginContext, key: string, value: ConfigValue) {
-  if (!value) {
-    return;
+// function checkAndSaveConfig(context: PluginContext, key: string, value: ConfigValue) {
```
Comment (swatDong) [inline packages/fx-core/src/plugins/resource/aad/utils/configs.ts:17]: Why comment out but not remove?

File: packages/fx-core/src/common/localSettingsConstants.ts (hunk 1 above)
Comment (dooriya) [reply packages/fx-core/src/common/localSettingsConstants.ts]: Agree. updated in the latest commit.

**Review by swatDong — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/aad/utils/configs.ts (hunk 2 above)
Comment (dooriya) [reply packages/fx-core/src/plugins/resource/aad/utils/configs.ts]: Just forgot to remove it . Already clean up.

**Review by KennethBWSong — APPROVED** [review]

**Review by qinezh — APPROVED** [review]

**Pull request merged**